app/Services/VersionService.php

```diff
@@ -90,15 +75,68 @@ public function publishNewVersionFromGithub(string $tag)
             throw new VersionAlreadyExistsException();
         }
 
+        try {
+            $normalisedVersion = $this->versionParser->normalize($tag);
+        } catch (UnexpectedValueException) {
+            Log::error(sprintf('Invalid release channel %s', $tag));
+            throw new ReleaseChannelNotFoundException();
+        }
+
+        $releaseChannel = PluginReleaseChannel::where('name', VersionParser::parseStability($normalisedVersion))
+            ->first();
+
+        if (!$releaseChannel) {
+            Log::error(sprintf('Invalid release channel %s', $tag));
+            throw new ReleaseChannelNotFoundException();
+        }
```

Comment:
Could be simplified by using `firstOrFail` and catching the `ModelNotFoundException`?

Reply:
Thought about this - I'm thinking to keep as we need the same exception class further down. Plus saves us having to catch another exception in the controller and Github gets grumpy if you respond with 404 to its web hooks :D

Reply:
Sound reasoning - cheers